### ISSUE TITLE
No messages, personal page and local toots after update to 3.4.3 (Error 500)

### ISSUE THREAD

**Issue by artemanufrij** [mastodon/mastodon#16999]
### Expected behaviour

Just work

### Actual behaviour

thows a lot of errors

### Steps to reproduce the problem

After update my instance from 3.4.1 to 3.4.3 I have a lot of problems:

## 1. Local toots
Local page throws 500 error:
![image](https://user-images.githubusercontent.com/16517095/142055220-34a4023a-468e-46cf-8b19-8da80f1e688a.png)

```
ActionController::UrlGenerationError (No route matches {:action=>"show", :controller=>"media", :id=>#<MediaAttachment status_id: 107157144501625384, file_file_name: "db2abdafcadaea6a.jpg", file_content_type: "image/jpeg", file_file_size: 379014, file_updated_at: "2021-10-24 15:08:44.965271000 +0000", remote_url: "", created_at: "2021-10-24 15:08:44.338546000 +0000", updated_at: "2021-10-24 15:11:04.526726000 +0000", shortcode: nil, type: "image", file_meta: {"original"=>{"width"=>1000, "height"=>486, "size"=>"1000x486", "aspect"=>2.05761316872428}, "small"=>{"width"=>574, "height"=>279, "size"=>"574x279", "aspect"=>2.057347670250896}}, account_id: 106767212766005419, id: 107157135316832376, description: nil, scheduled_status_id: nil, blurhash: "ULG94W~l%0%2_KtQj]s;IVM~ogW.R%WBo1oc", processing: "complete", file_storage_schema_version: 1, thumbnail_file_name: nil, thumbnail_content_type: nil, thumbnail_file_size: nil, thumbnail_updated_at: nil, thumbnail_remote_url: nil>}, possible unmatched constraints: [:id]
``` 

## 2. Messages are empty
![image](https://user-images.githubusercontent.com/16517095/142055373-93e4749b-6eef-4634-bb48-d009fe24d08b.png)

## 3. Profile page has an error
https://social.anufrij.de/@artem
![image](https://user-images.githubusercontent.com/16517095/142055487-bb00bcf3-7d75-4d6c-a52c-1def86a86a1a.png)

### Specifications

Mastodon Version: 3.4.3
I use docker-compose.yml on an arch linux for my instance.

**Comment by ClearlyClaire** [comment]
Did you, by any chance, accidentally switch to `main` before going back to 3.4.3? From the error, it seems one of your local media attachments was created using code from `main` rather than a release.

In `main`, we have stopped setting `shortcode` for new attachments and deprecated access by shortcode. But a change was needed for the server-side code to handle shortcode-less media attachments properly. That change is not included in 3.4.3, as 3.4.3 still sets shortcodes on new attachments.

To add a shortcode to existing media attachments that do not have one, run the following in a Rails console (`RAILS_ENV=production bundle exec rails c`):
```ruby
MediaAttachment.local.where(shortcode: nil).find_each { |m| m.shortcode = SecureRandom.urlsafe_base64(14) ; m.save! }
```

**Comment by artemanufrij** [comment]
@ClearlyClaire I love you :hugs: . It works!!!
